core/modules/system/system.install

```diff
@@ -2650,6 +2650,53 @@ function system_update_1045() {
 }
 
 /**
+ * Upgrade or prepare for transliteration functions in core.
+ */
+function system_update_1046() {
+  $config = config('system.core');
+
+  // If contrib module was installed, migrate settings and disable module.
+  if (module_exists('transliteration')) {
+    $contrib_config = config('transliteration.settings');
+    $config->set('transliteration_file_lowercase', $contrib_config->get('file_lowercase'));
+    $config->set('transliteration_file_uploads', $contrib_config->get('file_uploads'));
+    $config->set('transliteration_file_uploads_display_name', $contrib_config->get('file_uploads_display_name'));
+    config_set('search.settings', 'transliteration', $contrib_config->get('search'));
+
+    module_disable(array('transliteration'), FALSE);
+    $path = backdrop_get_path('module', 'transliteration');
+    backdrop_set_message(t('Backdrop core now provides transliteration functions in Core. A copy of Transliteration module has been located at %path. The module has been disbaled, you can uninstall and remove it safely from your installation.', array('%path' => BACKDROP_ROOT . '/' . $path)), 'warning');
+  }
+  else {
+    // Migrate variables to config.
+    $config->set('transliteration_file_lowercase', update_variable_get('transliteration_file_lowercase', TRUE));
+    $config->set('transliteration_file_uploads', update_variable_get('transliteration_file_uploads', TRUE));
+    $config->set('transliteration_file_uploads_display_name', update_variable_get('transliteration_file_uploads_display_name', TRUE));
+    config_set('search.settings', 'transliteration', update_variable_get('transliteration_search', TRUE));
+  
+    // Remove variables from D7.
+    update_variable_del('transliteration_file_lowercase');
+    update_variable_del('transliteration_file_uploads');
+    update_variable_del('transliteration_file_uploads_display_name');
+    update_variable_del('transliteration_search');
+
+    // Check for files that need to be fixed.
+    module_load_include('inc', 'system', 'system.admin');
+
+    if (!$query = transliteration_file_query()) {
+      // Database not supported.
+      return;
+    }
+    if (!$query->countQuery()->execute()->fetchColumn()) {
+      // Don't bother if no files need to be fixed.
+      return;
+    }
+    backdrop_set_message(t('Transliteration has been installed. <a href="@transliteration-url">Fix existing file names</a>.', array('@transliteration-url' => url('admin/config/media/file-system/transliteration'))));
```

Comment:
Maybe we could remove this entire message. We don't typically provide a message if the user does not need to take explicit action. If we'd like a message of some kind, I'd suggest returning this translated string rather than making it a message.